servant-examples/auth-combinator/auth-combinator.hs

```diff
@@ -56,25 +49,54 @@ newtype PublicData = PublicData { somedata :: Text }
 
 instance ToJSON PublicData
 
+-- | Our private API that we want to be auth-protected.
+type PrivateAPI = Get '[JSON] [PrivateData]
+
+-- | Our public API that doesn't have any protection
+type PublicAPI = Get '[JSON] [PublicData]
+
+-- | Our API, with auth-protection
+type API = "private" :> AuthProtect "cookie-auth" :> PrivateAPI
+      :<|> "public"  :> PublicAPI
+
+-- | A value holding our type-level API
 api :: Proxy API
 api = Proxy
 
+-- | We need to specify the data returned after authentication
+type instance AuthReturnType (AuthProtect "cookie-auth") = User
```

Comment:
@jkarni @soenkehahn I wouldn't mind getting some feedback on this design decision. The auth combinators don't carry any `-server` or `-client` specific data type in them. Therefore, we use the type family `AuthReturnType` to specify what would have usually been encoded as `usr`.

I ended up having to do this kind of thing to support `servant-client`. The previous datatype for auth, as an example, was: `AuthProtect (tag :: Symbol) usr` and when implementing the client I had to allow users to define a type they would use to add request data to the request. My two options were: use a type family to specify the data type or add something to `AuthProtect` like: `AuthProtect (tag :: symbol) clientData usr`

I ended up choosing the type family as it made for a cleaner API. But then I thought we should apply the same principal to `-server`.

I then removed any `usr` types from both `AuthProtect` and `BasicAuth`. Would love your thoughts on this.